docs/uml/ModelContainerConcept.puml

```diff
@@ -0,0 +1,69 @@
+@startuml
+Interface InputContainer {
+    + List<UniqueEntity> allEntitiesAsList()
+    + boolean validate()
+}
+
+Class RawGridElements {
+    - Set<NodeInput> nodes
+    - Set<LineInput> lines
+    - Set<Transformer2WInput> transformers2W
+    - Set<Transformer2WInput> transformers3W
+    - Set<SwitchInput> switches
+    - Set<MeasurementUnitInput> measurementUnits
+    + RawGridElements(attribute sets ...)
+    + RawGridElements(Collection<RawGridElements> rawGridElements)
+}
+InputContainer <|-- RawGridElements
+
+Class SystemParticipants {
+  - Set<BmInput> bmPlants
+  - Set<ChpInput> chpPlants
+  - Set<EvcsInput> evCS
+  - Set<FixedFeedInInput> fixedFeedIns
+  - Set<HpInput> heatPumps
+  - Set<LoadInput> loads
+  - Set<PvInput> pvPlants
+  - Set<StorageInput> storages
+  - Set<WecInput> wecPlants
+  + SystemParticipants(attribute sets ...)
+  + SystemParticipants(Collection<SystemParticipants> systemParticipants)
+}
+InputContainer <|-- SystemParticipants
+
+Class GraphicElements {
+  - Set<NodeGraphicInput> nodeGraphics
+  - Set<LineGraphicInput> lineGraphics
+  + GraphicElements(attribute sets ...)
+  + GraphicElements(Collection<GraphicElements> graphicElements)
+}
+InputContainer <|-- GraphicElements
+
+Abstract Class GridContainer {
+  - String gridName
+  - RawGridElements rawGrid
+  - SystemParticipants systemParticipants
+  - GraphicElements graphics
+  + GridContainer(String gridName, RawGridElements rawGrid, SystemParticipants systemParticipants, GraphicElements graphics)
+}
+InputContainer <|-- GridContainer
+RawGridElements *-- GridContainer
+SystemParticipants *-- GridContainer
+GraphicElements *-- GridContainer
+
+Class JointGridContainer {
+  - ImmutableGraph<SubGridContainer> subGridTopologyGraph
+  + JointGridContainer(attributes w\o graph)
+  + JointGridContainer(attributes w\o graph, ImmutableGraph<SubGridContainer> subGridTopologyGraph)
+  - boolean checkSubGridDependencyGraph(ImmutableGraph<SubGridContainer> subnetDependencyGraph)
+}
+GridContainer <|-- JointGridContainer
+
+Class SubGridContainer{
+  - int subnet
+  - VoltageLevel predominantVoltageLevel
+  + JointGridContainer(attributes w\o graph)
+  + JointGridContainer(attributes w\o graph, ImmutableGraph<SubGridContainer> subGridTopologyGraph)
+}
```

Comment:
This description does _not_ fit with the real implementation. 

As I don't know which one is correct I cannot resolve it.